pkg/clients/connectioncache/connectioncache_iface.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package connectioncache
+
+import (
+	"context"
+
+	"github.com/zeebo/errs"
+	"google.golang.org/grpc"
+	monkit "gopkg.in/spacemonkeygo/monkit.v2"
+
+	"storj.io/storj/pkg/overlay"
+	proto "storj.io/storj/protos/overlay"
+)
+
+var (
+	mon   = monkit.Package()
+	Error = errs.Class("error")
+)
+
+// Client indicates the supported clients
+// It can be any client type, with whom trying to establish connection.
+type Client int
+
+func (tc Client) String() string {
```

Comment:
i don't think you need this Client thing. grpc connections are grpc connections, independent of anything else